cloud/pkg/devicecontroller/controller/downstream.go

```diff
@@ -391,11 +392,39 @@ func (dc *DownstreamController) deviceAdded(device *v1alpha2.Device) {
 			klog.Errorf("Failed to send device addition message %v due to error %v", msg, err)
 		}
 
-		dc.sendDeviceModelMsg(device, model.InsertOperation)
+		if !isExistModel(&dc.deviceManager.Device, device) {
+			dc.sendDeviceModelMsg(device, model.InsertOperation)
+		}
 		dc.sendDeviceMsg(device, model.InsertOperation)
 	}
 }
 
+// isExistModel check if the target node already has the model.
+func isExistModel(deviceMap *sync.Map, device *v1alpha2.Device) bool {
+	var res bool
+	targetNode := device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions[0].Values[0]
+	modelName := device.Spec.DeviceModelRef.Name
+	deviceMap.Range(func(k, v interface{}) bool {
+		if k == device.Name {
```

Comment:
I suggest adding some annotation here.
// To find another device in deviceMap that uses the same deviceModel with exclude current device

Reply:
OK! I will make it.